### PR TITLE
main_v2 - fby3.5: cl: Support TMP431 sensor reading

### PR DESCRIPTION
**Pull request by DelphineChiu**
fby3.5: cl: Support TMP431 sensor reading
Summary:
- Support TMP431 sensor device
  For EVT3 ADI system, the "HSC Temp" and "MB Outlet Temp" should be read from TMP431 chip.
  For EVT3 MPS system, the "HSC Temp" sensor value is read from MP5990 and "MB Outlet Temp" is read from TMP75.
  
  Dependency: #230 

Test Plan:
1. Build code: pass
2. [EVT3 MPS] Check "HSC Temp" and "MB Outlet Temp" sensor reading: pass
3. [EVT3 ADI] Check "HSC Temp" and "MB Outlet Temp" sensor reading: pass
4. [POC] Check "HSC Temp" and "MB Outlet Temp" sensor reading: pass

Log:
1. Class type: class-1, 1ou present status: false, 2ou present status: true, board revision: EVT3(ADI)
root@bmc-oob:~# sensor-util slot1|grep "HSC\|MB Outlet Temp"
MB Outlet Temp               (0x2) :   25.44 C     | (ok)
HSC Temp                     (0xE) :   23.81 C     | (ok)
HSC Input Vol                (0x29) :   12.00 Volts | (ok)
HSC Output Cur               (0x30) :    0.28 Amps  | (ok)
HSC Input Pwr                (0x39) :    3.53 Watts | (ok)

2. Class type: class-1, 1ou present status: false, 2ou present status: false, board revision: EVT3(MPS)
root@bmc-oob:~# sensor-util slot1|grep "HSC\|MB Outlet Temp"
MB Outlet Temp               (0x2) :   24.00 C     | (ok)
HSC Temp                     (0xE) :   25.00 C     | (ok)
HSC Input Vol                (0x29) :   12.22 Volts | (ok)
HSC Output Cur               (0x30) :    0.25 Amps  | (ok)
HSC Input Pwr                (0x39) :    6.00 Watts | (ok)

3. Class type: class-1, 1ou present status: false, 2ou present status:false, board revision: POC
root@bmc-oob:~# sensor-util slot3|grep "HSC\|MB Outlet Temp"
MB Outlet Temp               (0x2) :   33.00 C     | (ok)
HSC Temp                     (0xE) :   29.52 C     | (ok)
HSC Input Vol                (0x29) :   12.00 Volts | (ok)
HSC Output Cur               (0x30) :    8.82 Amps  | (ok)
HSC Input Pwr                (0x39) :  108.70 Watts | (ok)

### CODE REVIEW

File: common/dev/tmp431.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#include "hal_i2c.h"
+#include "libutil.h"
+
+#define I2C_DATA_SIZE 5
+#define RANGE_0_127 0
+#define RANGE_m55_150 1
+
+static uint8_t temperature_range = 0xFF;
+
+uint8_t tmp431_read(uint8_t sensor_num, int *reading)
```
Comment (GoldenBug) [inline common/dev/tmp431.c:16]: Should we check to see if `sensor_num` is in a valid range?

0 <= sensor_num < MAX_NUM_SENSORS ?

File: common/dev/tmp431.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		return SENSOR_UNSPECIFIED_ERROR;
+	}
+
+	sensor_val *sval = (sensor_val *)reading;
+	sval->integer = (int32_t)val;
+	sval->fraction = (int32_t)(val * 1000) % 1000;
+	return SENSOR_READ_SUCCESS;
+}
+
+uint8_t tmp431_init(uint8_t sensor_num)
```
Comment (GoldenBug) [inline common/dev/tmp431.c:74]: Same, check if sensor_num is in valid range?

File: meta-facebook/yv35-cl/src/platform/plat_sensor_table.c [hunk 3]
```diff
@@ -242,6 +252,44 @@ void pal_fix_sensor_config()
...
+			 * For these two sensors, the reading values are read from TMP431 chip.data.num
+			 */
+			sensor_count = ARRAY_SIZE(evt3_class1_adi_temperature_sensor_table);
+			while (sensor_count != 0) {
+				add_sensor_config(
+					evt3_class1_adi_temperature_sensor_table[--sensor_count]);
+				if (sensor_count == 0) {
+					break;
+				}
+			}
```
Comment (GoldenBug) [inline meta-facebook/yv35-cl/src/platform/plat_sensor_table.c:292]: This seems like it might be better as a for loop?

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: common/dev/tmp431.c (hunk 1 above)
Comment (RenChen-wiwynn) [reply common/dev/tmp431.c:16]: We have fixed this and do the forced update.
The sensor_num is unsigned type so only judge if it's less than MAX_NUM_SENSORS. 
Please review it. Thanks.

File: common/dev/tmp431.c (hunk 2 above)
Comment (RenChen-wiwynn) [reply common/dev/tmp431.c:74]: We have fixed this and do the forced update.
The sensor_num is unsigned type so only judge if it's less than MAX_NUM_SENSORS. 
Please review it. Thanks.

File: meta-facebook/yv35-cl/src/platform/plat_sensor_table.c (hunk 3 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-cl/src/platform/plat_sensor_table.c]: We have fixed this and do the forced update.
Please review it. Thanks.

**Comment by DelphineChiu** [comment]
close the PR since the codes have been merged

**Pull request closed without merging**